src/lib/ngsi/Metadata.cpp

```diff
@@ -259,11 +259,14 @@ std::string Metadata::render(const std::string& indent, bool comma)
 
     if (compoundValueP->isObject())
     {
+      // Note in this case we don't add the "value" key, the toJson()
+      // method does it for toplevel compound (a bit crazy... this deserves a FIXME mark)
+      compoundValueP->renderName = true;
       part = compoundValueP->toJson(true, false);
     }
     else if (compoundValueP->isVector())
     {
-      part = "[" + compoundValueP->toJson(true, false) + "]";
+      part = JSON_STR("value") + ": [" + compoundValueP->toJson(true, false) + "]";
```

Comment:
I wonder how metadata rendering could work in the case of vector values... Maybe we don't have any test covering that.

It worth the effort to evaluate the vector metadata rendering comparing develop with this branch results in order to know if we have fixed an existing bug in the process.